libraries/aws_dynamodb_table.rb

```diff
@@ -12,63 +12,33 @@ class AwsDynamoDbTable < AwsResourceBase
     end
   "
 
-  attr_reader :table_name, :table_status, :creation_date, :number_of_decreases_today,
-              :write_capacity_units, :read_capacity_units, :item_count, :table_arn, :attributes, :key_schema, :global_secondary_indexes
-
   def initialize(opts = {})
     opts = { table_name: opts } if opts.is_a?(String)
     super(opts)
     validate_parameters(required: [:table_name])
+    @display_name = opts[:table_name]
 
     catch_aws_errors do
-      begin
-        resp = @aws.dynamodb_client.describe_table(table_name: opts[:table_name])
-        return nil if resp[0].nil? || resp[0].empty?
-      rescue Aws::DynamoDB::Errors::ResourceNotFoundException
-        return
-      end
-      @dynamodb_table = resp[0]
-      @table_name                = @dynamodb_table[:table_name]
-      @table_status              = @dynamodb_table[:table_status]
-      @table_arn                 = @dynamodb_table[:table_arn]
-      @creation_date             = @dynamodb_table[:creation_date_time].strftime('%m/%d/%Y')
-      @number_of_decreases_today = @dynamodb_table[:provisioned_throughput][:number_of_decreases_today]
-      @write_capacity_units      = @dynamodb_table[:provisioned_throughput][:write_capacity_units]
-      @read_capacity_units       = @dynamodb_table[:provisioned_throughput][:read_capacity_units]
-      @item_count                = @dynamodb_table[:item_count]
-      @attributes                = []
-      @key_schema                = []
-      @global_secondary_indexes  = []
-
-      @dynamodb_table[:attribute_definitions]&.map { |attr|
-        @attributes.push({ 'attribute_name': attr.attribute_name,
-                           'attribute_type': attr.attribute_type })
-      }
-      @dynamodb_table[:key_schema]&.map { |key|
-        @key_schema.push({ 'attribute_name': key.attribute_name,
-                           'key_type':       key.key_type })
-      }
-
-      if @dynamodb_table[:global_secondary_indexes]
-        gsi = { global_secondary_indexes: @dynamodb_table[:global_secondary_indexes] }
-        create_resource_methods(gsi)
-        global_secondary_indexes.each do |idx|
-          idx.define_singleton_method(:to_s) do
-            "AWS Dynamodb Global Secondary Index #{idx.item[:index_name]}"
-          end
-          idx.define_singleton_method(:key_schema) do
-            idx.item[:key_schema].map { |attr| { 'attribute_name': attr[:attribute_name], 'key_type': attr[:key_type] } }
-          end
-        end
```

Comment:
Is all of this information still available in the response when `create_resource_methods` is used?

Reply:
If we are removing some of these properties or if they are accessible in a different way, we should probably be updating the docs: https://github.com/inspec/inspec-aws/blob/526369afeebff32aac10315c10090d9f91f2b7ba/docs/resources/aws_dynamodb_table.md#properties

Reply:
Ended up reverting to how it was originally done and just extended it for the sse_description to figure out if it is encrypted.